webapp/components/generic/SearchableInput/SearchableInput.vue

```diff
@@ -0,0 +1,236 @@
+      delay: 300,
+    }
+  },
+  computed: {
+    emptyText() {
+      return this.isActive && !this.pending ? this.$t('search.failed') : this.$t('search.hint')
+    },
+    isActive() {
+      return !isEmpty(this.previousSearchTerm)
+    },
+  },
+  methods: {
+    isFirstOfType(option) {
+      return (
+        this.options.findIndex(o => o === option) ===
+        this.options.findIndex(o => o.__typename === option.__typename)
+      )
+    },
+    onFocus(event) {
+      clearTimeout(this.searchProcess)
+      this.isOpen = true
+    },
+    handleInput(event) {
+      clearTimeout(this.searchProcess)
+      this.value = event.target ? event.target.value.replace(/\s+/g, ' ').trim() : ''
+      this.isOpen = true
+      this.unprocessedSearchInput = this.value
+      if (isEmpty(this.value) || this.value.replace(/\s+/g, '').length < 3) {
+        return
+      }
+      this.searchProcess = setTimeout(() => {
+        this.previousSearchTerm = this.value
+        this.$emit('query', this.value)
+      }, this.delay)
+    },
+    /**
+     * TODO: on enter we should go to a dedicated search page!?
+     */
+    onEnter(event) {
+      this.isOpen = false
+      clearTimeout(this.searchProcess)
+      if (!this.pending) {
+        this.previousSearchTerm = this.unprocessedSearchInput
+        this.$emit('query', this.unprocessedSearchInput)
+      }
+    },
+    onDelete(event) {
+      clearTimeout(this.searchProcess)
+      const value = event.target ? event.target.value.trim() : ''
+      if (isEmpty(value)) {
+        this.clear()
+      } else {
+        this.handleInput(event)
+      }
+    },
+    clear() {
+      this.isOpen = false
+      this.unprocessedSearchInput = ''
+      this.previousSearchTerm = ''
+      this.searchValue = ''
+      this.$emit('clearSearch')
+      clearTimeout(this.searchProcess)
+    },
+    onBlur(event) {
+      this.searchValue = this.previousSearchTerm
+      this.isOpen = false
+      clearTimeout(this.searchProcess)
+    },
+    onSelect(item) {
+      this.isOpen = false
+      this.goToResource(item)
+      this.$nextTick(() => {
+        this.searchValue = this.previousSearchTerm
+      })
+    },
+    isPost(item) {
+      return item.__typename === 'Post'
+    },
+    goToResource(item) {
+      this.$nextTick(() => {
+        this.$router.push({
+          name: this.isPost(item) ? 'post-id-slug' : 'profile-id-slug',
+          params: { id: item.id, slug: item.slug },
+        })
+      })
+    },
+  },
+}
+</script>
+<style lang="scss" `>
+.searchable-input {
+  display: flex;
+  align-self: center;
+  width: 100%;
+  position: relative;
+  $padding-left: $space-x-small;
+  .search-heading {
+    display: flex;
+    flex-wrap: wrap;
+    font-weight: bold;
+    cursor: default;
+    background-color: white;
+    margin: -8px;
+    padding: 8px;
+  }
+  .extra-space {
+    margin-top: 8px;
+    padding-top: 4px;
+  }
+  .flex-span {
+    display: flex;
+    flex-wrap: wrap;
+  }
```

Comment:
These sound like utility-classes which is not the direction we decided to go in – could you come up with more descriptive names, such as `first-option` or `option-with-heading`?

Also, please use design tokens instead of 'magic numbers' anywhere in the CSS! :dizzy: